newIDE/app/src/AssetStore/BehaviorStore/index.js

```diff
@@ -0,0 +1,276 @@
+  const {
+    filters,
+    searchResults,
+    error,
+    fetchBehaviorsAndFilters,
+    filtersState,
+    searchText,
+    setSearchText,
+    allCategories,
+    chosenCategory,
+    setChosenCategory,
+    setInstalledBehaviorMetadataList,
+  } = React.useContext(BehaviorStoreContext);
+
+  const [showDeprecated, setShowDeprecated] = React.useState(false);
+
+  React.useEffect(
+    () => {
+      setInstalledBehaviorMetadataList(
+        showDeprecated
+          ? [
+              ...installedBehaviorMetadataList,
+              ...deprecatedBehaviorMetadataList,
+            ]
+          : installedBehaviorMetadataList
+      );
+    },
+    [
+      deprecatedBehaviorMetadataList,
+      installedBehaviorMetadataList,
+      setInstalledBehaviorMetadataList,
+      showDeprecated,
+    ]
+  );
+
+  React.useEffect(
+    () => {
+      fetchBehaviorsAndFilters();
+    },
+    [fetchBehaviorsAndFilters]
+  );
+
+  const filteredSearchResults = searchResults ? searchResults : null;
+
+  const tagsHandler = React.useMemo(
+    () => ({
+      add: filtersState.addFilter,
+      remove: filtersState.removeFilter,
+      chosenTags: filtersState.chosenFilters,
+    }),
+    [filtersState]
+  );
+
+  const getExtensionsMatches = React.useCallback(
+    (
+      extensionShortHeader: BehaviorShortHeader | SearchableBehaviorMetadata
+    ): SearchMatch[] => {
+      if (!searchResults) return [];
+      const extensionMatches = searchResults.find(
+        result => result.item.type === extensionShortHeader.type
+      );
+      return extensionMatches ? extensionMatches.matches : [];
+    },
+    [searchResults]
+  );
+
+  const { DismissableTutorialMessage } = useDismissableTutorialMessage(
+    'intro-behaviors-and-functions'
+  );
+
+  const showExtensionUpdateConfirmation = useExtensionUpdateAlertDialog();
+
+  const installAndChoose = React.useCallback(
+    async (
+      behaviorShortHeader: BehaviorShortHeader | SearchableBehaviorMetadata
+    ) => {
+      const isExtensionAlreadyInstalled =
+        behaviorShortHeader.extensionName &&
+        project.hasEventsFunctionsExtensionNamed(
+          behaviorShortHeader.extensionName
+        );
+      if (isExtensionAlreadyInstalled) {
+        const shouldUpdateExtension = await showExtensionUpdateConfirmation();
+        if (!shouldUpdateExtension) {
+          return;
+        }
+      }
+
+      if (behaviorShortHeader.url) {
+        sendExtensionAddedToProject(behaviorShortHeader.name);
+        const wasInstalled = await onInstall(behaviorShortHeader);
+        if (wasInstalled) {
+          onChoose(behaviorShortHeader.type);
+        } else {
+          // TODO Handle installation failures.
+        }
+      } else {
+        onChoose(behaviorShortHeader.type);
+      }
+    },
+    [project, onChoose, showExtensionUpdateConfirmation, onInstall]
+  );
+
+  return (
+    <React.Fragment>
+      <ResponsiveWindowMeasurer>
```

Comment:
Use `useResponsiveWindowWidth` to avoid 2 extra indentation.

Reply:
Actually, it seems that `windowWidth` is not used (same for extensions).